docs/source/common/lightning_module.rst

```diff
@@ -482,118 +509,76 @@ There are two ways to call ``test()``:
-            self.decoder = nn.Sequential(nn.Linear(latent_dim, 256), nn.ReLU(), nn.Linear(256, 28 * 28))
-
-        def training_step(self, batch, batch_idx):
-            x, _ = batch
-
-            # encode
-            x = x.view(x.size(0), -1)
-            z = self.encoder(x)
+.. note::
 
-            # decode
-            recons = self.decoder(z)
+    It is recommended to test on single device since Distributed Training such as DDP internally
+    uses :class:`~torch.utils.data.distributed.DistributedSampler` which replicates some samples to
+    make sure all devices have same batch size in case of uneven inputs. This is helpful to make sure
+    benchmarking for research papers is done the right way.
 
-            # reconstruction
-            reconstruction_loss = nn.functional.mse_loss(recons, x)
-            return reconstruction_loss
-
-        def validation_step(self, batch, batch_idx):
-            x, _ = batch
-            x = x.view(x.size(0), -1)
-            z = self.encoder(x)
-            recons = self.decoder(z)
-            reconstruction_loss = nn.functional.mse_loss(recons, x)
-            self.log("val_reconstruction", reconstruction_loss)
+----------
 
-        def predict_step(self, batch, batch_idx, dataloader_idx=0):
-            x, _ = batch
+*********
+Inference
+*********
 
-            # encode
-            # for predictions, we could return the embedding or the reconstruction or both based on our need.
-            x = x.view(x.size(0), -1)
-            return self.encoder(x)
+Prediction Loop
+===============
 
-        def configure_optimizers(self):
-            return torch.optim.Adam(self.parameters(), lr=0.0002)
+By default, the :meth:`~pytorch_lightning.core.lightning.LightningModule.predict_step` method runs the
+:meth:`~pytorch_lightning.core.lightning.LightningModule.forward` method. In order to customize this behaviour,
+simply override the :meth:`~pytorch_lightning.core.lightning.LightningModule.predict_step` method.
 
-Which can be trained like this:
+For the example let's override ``predict_step`` and try out `Monte Carlo Dropout <https://arxiv.org/pdf/1506.02142.pdf>`_:
 
 .. code-block:: python
 
-    autoencoder = Autoencoder()
-    trainer = pl.Trainer(gpus=1)
-    trainer.fit(autoencoder, train_dataloader, val_dataloader)
-
-This simple model generates examples that look like this (the encoders and decoders are too weak)
-
-.. figure:: https://pl-bolts-doc-images.s3.us-east-2.amazonaws.com/pl_docs/ae_docs.png
-    :width: 300
+    class LitModel(pl.LightningModule):
+        def __init__(self):
+            super().__init__()
+            self.linear = nn.Linear(4, 5)
+            self.dropout = nn.Dropout()
 
-The methods above are part of the LightningModule interface:
+        def predict_step(self, batch, batch_idx):
+            # enable Monte Carlo Dropout
+            self.dropout.train()
```

Comment:
```suggestion
            self.train()
```

Reply:
dropout is the only one that needs to be kept in training mode.